### ISSUE TITLE
Plugin cause sometimes "ReferenceError: Can't find variable: cordova" ?

### ISSUE THREAD

**Issue by Bnaya** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#243]
I'm investigation js error on my cordova app that throned to some of the users.
I'm gathering js errors informations window error handler, i don't have access to those users devices and i've yet didn't manage to repro it myself

My code base dose not contains  any access to the cordova variable.
I've went through my plugins and all of them are expositing them selfs using  

``` js
module.exports = FileSystem;
```

That plugin is the only one that (tries to) access directly to cordova 

``` js
cordova.addConstructor(SocialSharing.install);
```

Not in one of the methods but in the main js file.

I don't deeply know the cordova plugins internal but i suspect that is some cases that code is  executed  before the cordova global variable is available, whats cause the error.

Other plugins also creating local cordova variable with:

``` js
cordova = require('cordova'), 
```

And this one doesn't.

This is very bad error, that i think can break the entire app.

Thanks!

**Comment by ccorcos** [comment]
I'm having this issue as well. Sometimes the plugins don't seem to work...

**Comment by leob** [comment]
I got something similar, but not related to the plugin specifically.

My Ionic app running on iOS 8 started to throw "Can't find variable: cordova" errors all over the place. No idea what could have caused it except for a Cordova upgrade.

Also no idea how to fix it, until after a lot of Googling I ran across this thread. I just tried a variant on Bnaya's suggestion, well actually I just stuffed this into my index.html:

  if (!window.cordova) {
    window.cordova = require('cordova');
  }

And, guess what, lo and behold, my iOS app started working again. Big scary workaround but well if it works it works ... stuff like this is not the fun part of being a mobile app dev.
